### DISCUSSION TITLE
Plex Scrobbler - Watching events in background on Windows

### DISCUSSION THREAD

**Discussion by saadats** [Taxel/PlexTraktSync#332] [Q&A]
> _Scrobbling simply means automatically tracking what you’re watching. Instead
> of checking in from your phone of the website, this command runs in the
> background and automatically scrobbles back to Trakt while you enjoy watching
> your media_ - [Plex Scrobbler@blog.trakt.tv][plex-scrobbler]

[plex-scrobbler]: https://blog.trakt.tv/plex-scrobbler-52db9b016ead

```shell
$ ./plex_trakt_sync.sh watch
```

Can anybody help me with this part. I've managed to sync properly on a Windows PC. However, syncing takes a lot of time and is inconvenient, if done on regular basis. How do I execute this command on a PC? I've done this thing before, but I've forgotten. Any help will really be appreciated.

**Comment by simonc56** [comment]
The `watch` command is a real-time listening of Plex events transmitted instantly toward Trakt. By definition it never ends listening until you stop it.

The `sync` command is a one-time sync of Plex to Trakt and Trakt to Plex based on the config file. The script scans the whole library and each difference found between Plex and Trakt is updated to keep items in the same state (watched, collected,...). This command takes a few seconds or minutes and have to be run periodically to keep Plex & Trakt up-to-date.

Personnally, I just use the `sync ` command every 12 hours.
Be aware that the first sync run takes a lot of time because the script has to build the local cache. Next runs are faster.

You can read again the [README](https://github.com/Taxel/PlexTraktSync#readme) file.

**Reply by saadats** [reply]
> The `watch` command is a real-time listening of Plex events transmitted instantly toward Trakt. By definition it never ends listening until you stop it.
> 
> The `sync` command is a one-time sync of Plex to Trakt and Trakt to Plex based on the config file. The script scans the whole library and each difference found between Plex and Trakt is updated to keep items in the same state (watched, collected,...). This command takes a few seconds or minutes and have to be run periodically to keep Plex & Trakt up-to-date.
> 
> Personnally, I just use the `sync ` command every 12 hours.
> Be aware that the first sync run takes a lot of time because the script has to build the local cache. Next runs are faster.
> 
> You can read again the [README](https://github.com/Taxel/PlexTraktSync#readme) file.

Thanks for your reply. My question was about how to keep the watch event running in the background on a Windows PC. The normal sync seems to be taking a lot of time, everytime I run it. Thanks.

Saadat

**Reply by dachewster99** [reply]
any updates on keeping this running in the background? I would prefer not to keep a terminal open at all times

**Reply by simonc56** [reply]
You can run it through a vbs script : https://serverfault.com/questions/9038/run-a-bat-file-in-a-scheduled-task-without-a-window

**Reply by dachewster99** [reply]
Thanks for the speedy reply. I ended up getting it to work just by adding a task with Task Scheduler, and setting it o run whether user is logged on or not

![image](https://user-images.githubusercontent.com/35871585/154666597-44643939-9983-40ba-899e-de6a0309ac47.png)
![image](https://user-images.githubusercontent.com/35871585/154666761-0e55919f-40b5-4922-aefd-8066a0606482.png)
![image](https://user-images.githubusercontent.com/35871585/154666798-947e78ed-68b3-40af-b2ee-7340c9d91225.png)

**Comment by simonc56** [comment]
There is no need to manually add a task because the setup script does it for you :

https://github.com/Taxel/PlexTraktSync/blob/781fe5ccf0ab5022bad3c07ad2903ce26a74218c/setup.bat#L66-L69

edit : this thread is about watch plex events in Windows background. Nothing else.

**Reply by dachewster99** [reply]
I didnt run the setup, I installed it via the `pipx` technique, and it didnt ever ask me to do that. I know when I had installed it before with the `setup.bat` technique, it did ask that, but the `pipx` method did not

**Reply by dachewster99** [reply]
Also that is just a daily sync, not the `watch`.

**Reply by simonc56** [reply]
Ok, then you can now :
- mark as answered
- send a PR to improve doc and help others in the same situation.

**Reply by dachewster99** [reply]
I would love to mark it as answered, but as I wasnt the OP, I dont think I have that option. Im not seeing it anywhere at least
